test/realm_value_test.dart

```diff
@@ -0,0 +1,158 @@
+
+    for (final x in values) {
+      test('Roundtrip ${x.runtimeType} $x', () {
+        final something = realm.write(() => realm.add(AnythingGoes(oneAny: RealmValue.from(x))));
+        expect(something.oneAny.value, x);
+        expect(something.oneAny, RealmValue.from(x));
+      });
+    }
+
+    test('Illegal value', () {
+      expect(() => realm.write(() => realm.add(AnythingGoes(oneAny: RealmValue.from(<int>[1, 2])))), throwsArgumentError);
+    });
+
+    for (final x in values) {
+      test('Switch $x', () {
+        final something = AnythingGoes(oneAny: RealmValue.from(x));
+        final value = something.oneAny.value;
+        switch (something.oneAny.type) {
+          case Null:
+            expect(value, isA<void>());
+            break;
+          case bool:
+            expect(value, isA<bool>());
+            break;
+          case String:
+            expect(value, isA<String>());
+            break;
+          case int:
+            expect(value, isA<int>());
+            break;
+          case double:
+            expect(value, isA<double>());
+            break;
+          case AnythingGoes: // RealmObject won't work with switch
+            expect(value, isA<AnythingGoes>());
+            expect(value, isA<RealmObject>());
+            break;
+          case DateTime:
+            expect(value, isA<DateTime>());
+            break;
+          case ObjectId:
+            expect(value, isA<ObjectId>());
+            break;
+          case Uuid:
+            expect(value, isA<Uuid>());
+            break;
+          default:
+            fail('${something.oneAny} not handled correctly in switch');
+        }
+      });
+    }
+
+    for (final x in values) {
+      test('If-is $x', () {
+        final something = AnythingGoes(oneAny: RealmValue.from(x));
+        final value = something.oneAny.value;
+        final type = something.oneAny.type;
+        if (value == null) {
+          expect(type, Null);
+        } else if (value is int) {
+          expect(type, int);
+        } else if (value is String) {
+          expect(type, String);
+        } else if (value is bool) {
+          expect(type, bool);
+        } else if (value is double) {
+          expect(type, double);
+        } else if (value is DateTime) {
+          expect(type, DateTime);
+        } else if (value is Uuid) {
+          expect(type, Uuid);
+        } else if (value is ObjectId) {
+          expect(type, ObjectId);
+        } else if (value is AnythingGoes) {
+          expect(type, AnythingGoes);
+        } else {
+          fail('$value not handled correctly in if-is');
+        }
+      });
+    }
+  });
+
+  group('List<RealmValue>', () {
+    final now = DateTime.now().toUtc();
+    final values = <Object?>[
+      null,
+      true,
+      'text',
+      42,
+      3.14,
+      AnythingGoes(),
+      now,
+      ObjectId.fromTimestamp(now),
+      Uuid.v4(),
+    ];
+
+    final config = Configuration.inMemory([AnythingGoes.schema]);
```

Comment:
We are testing only with inMemory Realm. Is that enough?

Reply:
Yes, I don't see a reason test with other configurations.

Reply:
> Okay, I had forgotten some things in dart, so I believe there's more work needed to support RealmValue, but that can probably be split into a different PR...

I will just comment that even trying to read a mixed property from core containing a realm object with unknown schema will crash, and since the property is mixed we really have no way to detect this, so the crash is inevitable 😢